streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -63,13 +63,15 @@ public void makeReady(final Map<InternalTopicConfig, Integer> topics) {
                 final MetadataResponse metadata = streamsKafkaClient.fetchMetadata();
                 final Map<String, Integer> existingTopicPartitions = fetchExistingPartitionCountByTopic(metadata);
                 final Map<InternalTopicConfig, Integer> topicsToBeCreated = validateTopicPartitions(topics, existingTopicPartitions);
-                if (metadata.brokers().size() < replicationFactor) {
-                    throw new StreamsException("Found only " + metadata.brokers().size() + " brokers, " +
-                        " but replication factor is " + replicationFactor + "." +
-                        " Decrease replication factor for internal topics via StreamsConfig parameter \"replication.factor\""  +
-                        " or add more brokers to your cluster.");
+                if (topicsToBeCreated.size() > 0) {
```

Comment:
this could be:
`if (!topicsToBeCreated.isEmpty() && metadata.brokers().size() < replicationFactor) {..}'